src/Gaffer/Monitor.cpp

```diff
@@ -66,7 +66,15 @@ Monitor::Scope::Scope( const MonitorPtr &monitor, bool active )
 	{
 		m_monitors.erase( monitor );
 	}
+
+	bool mightDisallowCacheRead = false;
+	for( const auto &m : m_monitors )
+	{
+		mightDisallowCacheRead |= m->mightDisallowCacheRead( ThreadState::CacheType::Value );
+		mightDisallowCacheRead |= m->mightDisallowCacheRead( ThreadState::CacheType::Hash );
```

Comment:
This is a bit problematic for my "open ended process types" world view. But I think it's probably OK to remove the `type` argument from `mightDisallowCacheRead()`?